pyqldb/errors/__init__.py

```diff
@@ -54,78 +57,101 @@ def __init__(self, error):
     """
     Is the exception an InvalidSessionException?
 
-    :type e: :py:class:`botocore.exceptions.ClientError`
-    :param e: The ClientError caught.
+    :type e: :py:class:`builtins.Exception`
+    :param e: The Exception caught.
 
     :rtype: bool
     :return: True if the exception is an InvalidSessionException. False otherwise.
     """
-    is_invalid_session = e.response['Error']['Code'] == 'InvalidSessionException'
-    return is_invalid_session
+    if isinstance(e, ClientError):
+        is_invalid_session = e.response['Error']['Code'] == 'InvalidSessionException'
+        return is_invalid_session
+    return False
 
 
 def is_transaction_expired_exception(e):
     """
     Does this exception denote that a transaction has expired?
 
-    :type e: :py:class:`botocore.exceptions.ClientError`
-    :param e: The ClientError caught.
+    :type e: :py:class:`builtins.Exception`
+    :param e: The Exception caught.
 
     :rtype: bool
     :return: True if the exception denote that a transaction has expired. False otherwise.
     """
-    is_invalid_session = e.response['Error']['Code'] == 'InvalidSessionException'
-
-    if "Message" in e.response["Error"]:
-        return is_invalid_session and re.search("Transaction .* has expired", e.response["Error"]["Message"])
+    # TODO CHECK IF THIS IS ClientError
+    if isinstance(e, ClientError):
+        is_invalid_session = e.response['Error']['Code'] == 'InvalidSessionException'
 
+        if "Message" in e.response["Error"]:
+            return is_invalid_session and re.search("Transaction .* has expired", e.response["Error"]["Message"])
     return False
 
 
 def is_retriable_exception(e):
     """
-    Is the exception a retriable exception?
+    Is the exception a retryable exception?
 
-    :type e: :py:class:`botocore.exceptions.ClientError`
-    :param e: The ClientError caught.
+    :type e: :py:class:`builtins.Exception`
+    :param e: The Exception caught.
 
     :rtype: bool
-    :return: True if the exception is a retriable exception. False otherwise.
+    :return: True if the exception is a retryable exception. False otherwise.
     """
-    is_retriable = e.response['ResponseMetadata']['HTTPStatusCode'] == 500 or \
-                   e.response['ResponseMetadata']['HTTPStatusCode'] == 503 or \
-                   e.response['Error']['Code'] == 'NoHttpResponseException' or \
-                   e.response['Error']['Code'] == 'SocketTimeoutException'
+
+    is_retriable = (isinstance(e, ClientError) and (e.response['ResponseMetadata']['HTTPStatusCode'] == 500 or
+                                                   e.response['ResponseMetadata']['HTTPStatusCode'] == 503 or
+                                                   e.response['Error']['Code'] == 'NoHttpResponseException' or
+                                                   e.response['Error']['Code'] == 'SocketTimeoutException')) or \
+                   isinstance(e, ReadTimeoutError) or \
```

Comment:
Seems like we added a whole bunch of items to this - more than I would have expected?

Reply:
In Python we also retry on HTTP retryable errors. I cleaned this up since `isinstance()` takes in a tuple